### DISCUSSION TITLE
JPA Entity / package local

### DISCUSSION THREAD

**Discussion by khmarbaise** [mapstruct/mapstruct#3614] [Q&A]
Currently I have serveral entities within their own packages associated with their services etc. that means having an entity similar to this:
```java
@Entity
class Employee {
  ...
  private String firstName;
  private String surname;

  String getFirstName() {
    return firstName;
  }

  void setFirstName(String name) {
    this.firstName = name;
  }

  String getSurname() {
    return surname;
  }

  void setSurname(String surname) {
    this.surname = surname;
  }

  EmployeeType getType() {
    return type;
  }

  void setType(EmployeeType type) {
    this.type = type;
  }

}
```
also having a DTO (record type) for simplicity with the same attributes and defined a mapping interface.

But what happens is that no callings of the JPA entity properties are made. Only if I make the properties public it will work. Is this intentionally? (If you like I can prepare a more complete example):

**Comment by filiphr** [comment]
Hey @khmarbaise, I think that I understand what is happening. We are currently only working using public properties. We have #1689 which I believe is what you need. The issue talks about fields, but it is the same for getters and setters as well.

**Reply by khmarbaise** [reply]
Hi @filiphr That's exactly my assumption, cause I tested with public made properties (getter/setter methods) and that works..
